### DISCUSSION TITLE
Working with rotated grids

### DISCUSSION THREAD

**Discussion by soizicg** [OpenDrift/opendrift#1373] [Q&A] [answered]
I use a ROMS forcing with a rotated grid, where a variable "angle" keeps track of the grid's rotation for each hourly output.
My question is about how a particle tracking model like OceanDrift handles these grids. In the "reader_ROMS_native.py" script, I can see that the velocity fields "x_sea_water_velocity" and "y_sea_water_velocity" are rotated to convert them into proper zonal and meridional velocities before computing particle trajectories.
However, I am confused about the necessity of this step. Isn't the particle tracking model using the original rotated grid for the particle trajectories? Then why is it necessary to convert the ROMS velocity output, which are rotated to the local coordinate space, into zonal and meridional velocities?
Thanks for the help.

**Comment by knutfrode** [comment, marked as answer]
Hi,

Internally, OpenDrift always uses vector components along east/north. Thus
it is the task of any reader to rotate vectors from a rotated/alternative
coordinate system to these directions, if necessary.

Knut-Frode

**Reply by soizicg** [reply]
Thanks for the quick answer
